pkg/envoy/ads/stream.go

```diff
@@ -20,6 +22,42 @@ import (
 	"github.com/openservicemesh/osm/pkg/utils"
 )
 
+// This is the workqueue Job implementation to issue a Send Response to a proxy
+type sendJob struct {
+	// Send Response parametres
+	typeurls  mapset.Set
+	proxy     *envoy.Proxy
+	cfg       configurator.Configurator
+	adsStream *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer
+	request   *xds_discovery.DiscoveryRequest
+	xdsServer *Server
+
+	// Optional waiter
+	jobDoneChannel chan struct{}
+}
```

Comment:
I'd recommend we create a `NewJob()` function in `pkg/envoy/ads/jobs.go` (filename as an example - I don't think it exists) and use that to initialize this struct.

This `NewJob()` could be a function receiver on `Server` even... unless that complicates things.
But NewJob can hide inside the creation of the `done` channel.